### ISSUE TITLE
Prepare branches for 8.x and 9.0 releases

### ISSUE THREAD

**Issue by simitt** [elastic/apm-server#13968]
In anticipation of a major version upgrade to `9.0` , following changes need to be made to the branching strategy:

```[tasklist]
### Initial setup for `9.0`
- [ ] branch out an `8.x` branch from `main` with version `8.16`
- [ ] change version in `main` branch to `9.0`
- [ ] automatically add `backport-8.x` label to all PRs against `main` branch, as any changes will likely go into `main` and `8.x` for a certain time (if label is manually removed, then no backport PR should be created). 
```

```[tasklist]
### Feature Freeze for `8.16` 
- [ ] branch out an `8.16` branch from `8.x`
- [ ] change version in `8.x` branch to `8.17`
- [ ] adapt to automatically add `backport-8.17` label to all PRs against `main` branch, instead of `backport-8.x` label
```

```[tasklist]
### Feature Freeze for `9.0` 
- [ ] remove automatic backport label to `8.17`
```

**Comment by v1v** [comment]
I created an initial PR with some of the requirements mentioned above: 
- https://github.com/elastic/apm-server/pull/13978

We, robots, are also tracking this effort in an internal GH issue. Glad we can collaborate on this together.

**Comment by simitt** [comment]
@v1v does the robots team have capacity to take care of the initial setup for `9.0` or should we plan for this? We need the branches to be ready by Sep 11th.

**Comment by v1v** [comment]
> @v1v does the robots team have capacity to take care of the initial setup for 9.0 or should we plan for this? We need the branches to be ready by Sep 11th.

Yes, we can work on it - it's planned in our current and upcoming iteration.

**Comment by simitt** [comment]
@v1v I updated the description above to the latest changes to include `8.17` and `8.18` correctly.

**Comment by raultorrecilla** [comment]
moving this to it-105

**Comment by simitt** [comment]
@v1v with the introduction of an `8.18` release in sync with the `9.0` release and another release `8.19` afterwards (in sync with `9.1`), we also need to adapt the automation listed in this task. I updated the description above, please reach out if anything is unclear.

**Comment by raultorrecilla** [comment]
in progress, moving it from it105 to it106

**Comment by v1v** [comment]
Some thoughts about the tasks in `Feature Freeze for 8.18, 9.0 and 8.19`:

- branch out an `8.18` branch from `8.x` - Automated 🤖  (see [run-minor-releases])
- change version in `8.x` branch to `8.19` - Automated 🤖  (see [run-minor-releases])
- remove automatic backport label to `8.x` - Manual ✋ 
- create label `backport-8.18`. - Automated 🤖   (see [run-minor-releases])
- create label `backport-8.19`. - Automated 🤖   (see [run-minor-releases])
- rename `8.x` branch to `8.19` - Manual ✋

In addition, I'm working on supporting major release automation in `main` for `9.0.0`, see https://github.com/elastic/apm-server/pull/15329


[run-minor-releases]: https://github.com/elastic/apm-server/blob/main/dev_docs/RELEASES.md#day-after-feature-freeze

**Comment by raultorrecilla** [comment]
moving it from it105 to it106

**Comment by kruskall** [comment]
This has been completed. Closing.
